### PR TITLE
chore: Loading data from unified GraphQL schema

### PR DESCRIPTION
**Pull request by paul-vd**
## What?
This documents how to load dta rom unified GraphQL schema.

## Example
- https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/merge_requests/2944

## Preview
- https://deploy-preview-798--heuristic-almeida-1a1f35.netlify.app/docs/remixed/get-started/loading-data-from-unified-graphql-schema

### CODE REVIEW

**Comment by real34** [comment]
**issue:** there are several incorrect things regarding this patch

> This updates the Your first route documentation to follow the example of hello-world.tsx provided in the skeleton. It also includes examples of how to load data from the unified graphql schema.

This is not correct. The issue was about adding a new page : "Get Started > Loading data from Front-Commerce's unified GraphQL schema"

The "Your first route" was already good enough for its goal.

Also, the example in the "advanced" route with fetching isn't realistic. The goal is to reproduce the same patterns that developers will find in FC. In the current state it isn't really close to what's in the core:
- in FC the query is loaded from `~/graphql/graphql`
- in FC we always have metas
- it could in fact (that wasn't in acceptance criteria) be better to show a page with a param (e.g: `:id`) that would allow us to show how to do error management (404)

The `_main.wishlist.share.$shareCode.($path).($itemId).tsx` route could be used as an inspiration for the most common patterns developers will see on a "dynamic page" that fetches data from GraphQL.

**Review by real34 — CHANGES_REQUESTED** [review]
**issue:** there are several incorrect things regarding this patch

> This updates the Your first route documentation to follow the example of hello-world.tsx provided in the skeleton. It also includes examples of how to load data from the unified graphql schema.

This is not correct. The issue was about adding a new page : "Get Started > Loading data from Front-Commerce's unified GraphQL schema"

The "Your first route" was already good enough for its goal.

Also, the example in the "advanced" route with fetching isn't realistic. The goal is to reproduce the same patterns that developers will find in FC. In the current state it isn't really close to what's in the core:
- in FC the query is loaded from `~/graphql/graphql`
- in FC we always have metas
- it could in fact (that wasn't in acceptance criteria) be better to show a page with a param (e.g: `:id`) that would allow us to show how to do error management (404)

The `_main.wishlist.share.$shareCode.($path).($itemId).tsx` route could be used as an inspiration for the most common patterns developers will see on a "dynamic page" that fetches data from GraphQL.

File: docs/01-get-started/loading-data-from-unified-graphql-schema.mdx [hunk 1]
```diff
@@ -0,0 +1,135 @@
+---
+title: Loading data from unified GraphQL schema
+description:
+  Explore the integration of a unified GraphQL schema in a Front-Commerce route
+  using Remix. This guide delves into the practical aspects of querying and
```
Comment (real34) [inline docs/01-get-started/loading-data-from-unified-graphql-schema.mdx:5]: **nitpick:** the sentence isn't very clear that it's about the app's unified schema exposed by FC

**suggestion:**

```suggestion
  Explore how to retrieve dynamic data from your application's unified GraphQL schema in a Front-Commerce route
  using Remix. This guide delves into the practical aspects of querying and
```

File: docs/01-get-started/loading-data-from-unified-graphql-schema.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+<p>{frontMatter.description}</p>
+
+## Prerequisites
+
+Before diving into the code, ensure you have the following prerequisites in
+place:
+
+- A Front-Commerce project configured and ready to use
+- Basic knowledge of
+  [Remix routes](https://remix.run/docs/en/main/discussion/routes)
```
Comment (real34) [inline docs/01-get-started/loading-data-from-unified-graphql-schema.mdx:22]: **nitpick:** I think it's better to link to our own content explaining the previous steps

**suggestion:** replace the link with our how-to guide explaining routes basics (it itself links to further official / advanced resources)

```suggestion
- Basic knowledge of
  [Remix routes](/docs/3.x/get-started/your-first-route)
```

File: docs/01-get-started/loading-data-from-unified-graphql-schema.mdx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+- [FAQ GraphQL Module](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/tree/c317ab057ae0927e294a5287eb8e0be73ca7e1fb/skeleton/extensions/faq-demo/graphql)
+  installed and configured
```
Comment (real34) [inline docs/01-get-started/loading-data-from-unified-graphql-schema.mdx:24]: **nitpick:** this is only useful for using the code as is

**suggestion:** 

```suggestion
- [FAQ GraphQL Module](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/tree/c317ab057ae0927e294a5287eb8e0be73ca7e1fb/skeleton/extensions/faq-demo/graphql)
  installed and configured if you want to execute code samples without any changes
```

File: docs/01-get-started/loading-data-from-unified-graphql-schema.mdx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    },
+  });
+  if (!response.voteIsFaqUseful) {
+    throw new Response("Question not found", { status: 404 });
+  }
+
+  return json({
+    success: response.voteIsFaqUseful?.success ?? false,
+  });
+};
```
Comment (real34) [inline docs/01-get-started/loading-data-from-unified-graphql-schema.mdx:87]: **issue:** this is part of the "Loading data" section which is incorrect

**suggestion:** create another "Mutate data" section explaining that same concepts seen for loading data can be used for mutating data

File: docs/01-get-started/loading-data-from-unified-graphql-schema.mdx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+This is an example of a route that fetches a question from the GraphQL API and
+renders it using a <FaqDetail /> Page component (that is a plain React
+component). As you can see, all the logic belongs to the route!
+
+## Reference
+
+As we can see here it's quite simple to create a new route and load data from
+external API's, but this only scratches the surface of what you can do with
+Front-Commerce and Remix.
```
Comment (real34) [inline docs/01-get-started/loading-data-from-unified-graphql-schema.mdx:117]:
```suggestion
As we can see here retrieving dynamic data from Front-Commerce unified GraphQL schema 
can be done in a natural way from any route. But this only scratches the surface of what you can do with
Front-Commerce and Remix.
```

**Review by Kagaminara — APPROVED** [review]
LGTM

File: docs/01-get-started/loading-data-from-unified-graphql-schema.mdx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  const error = useRouteError();
+
+  if (isRouteErrorResponse(error)) {
+    return <div>FAQ : question not found</div>;
+  }
+};
+```
+
+This is an example of a route that fetches a question from the GraphQL API and
+renders it using a <FaqDetail /> Page component (that is a plain React
```
Comment (real34) [inline docs/01-get-started/loading-data-from-unified-graphql-schema.mdx:83]: **issue:** the component is interpreted



**suggestion:** 

```suggestion
renders it using a `<FaqDetail />` Page component (that is a plain React
```

**Review by real34 — APPROVED** [review]

**Pull request merged**